src/boot/efi/util.c

```diff
@@ -91,10 +91,36 @@ EFI_STATUS efivar_set(const EFI_GUID *vendor, const CHAR16 *name, const CHAR16 *
 }
 
 EFI_STATUS efivar_set_uint(const EFI_GUID *vendor, CHAR16 *name, UINTN i, BOOLEAN persistent) {
-        CHAR16 str[32];
+        assert_cc(sizeof(UINTN) == sizeof(UINT64) || sizeof(UINTN) == sizeof(UINT32));
+
+        return sizeof(UINTN) == sizeof(UINT32) ? efivar_set_uint32(vendor, name, i, persistent) :
+                                                 efivar_set_uint64(vendor, name, i, persistent);
```

Comment:
this should better be done with #ifdeffery, i.e. strictla at compile time

Reply:
I did some searching but `sizeof()` isn't available in ifdefs. Any hints on how to do this at compile time?

Reply:
meson is your friend

Reply:
i.e. have a look at SIZEOF_TIME_T in meson.build.

Reply:
Not necessary anymore with the latest changes